pkg/combustion/kubernetes_integration_test.go

```diff
@@ -0,0 +1,173 @@
+//go:build integration
+
+package combustion
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/suse-edge/edge-image-builder/pkg/fileio"
+	"github.com/suse-edge/edge-image-builder/pkg/image"
+	"gopkg.in/yaml.v3"
+)
+
+func TestConfigureManifestsValidDownload(t *testing.T) {
+	// Setup
+	ctx, teardown := setupContext(t)
+	defer teardown()
+
+	ctx.ImageDefinition.Kubernetes.Manifests.URLs = []string{
+		"https://k8s.io/examples/application/nginx-app.yaml",
+	}
+
+	k8sCombDir := filepath.Join(ctx.CombustionDir, k8sDir)
+	require.NoError(t, os.Mkdir(k8sCombDir, os.ModePerm))
+
+	downloadedManifestsPath := filepath.Join(k8sDir, manifestsDir)
+	downloadedManifestsDestDir := filepath.Join(k8sCombDir, manifestsDir)
+	expectedDownloadedFilePath := filepath.Join(downloadedManifestsDestDir, "dl-manifest-1.yaml")
+
+	// Test
+	manifestsPath, err := configureManifests(ctx)
+
+	// Verify
+	require.NoError(t, err)
+	assert.Equal(t, downloadedManifestsPath, manifestsPath)
+
+	assert.FileExists(t, expectedDownloadedFilePath)
+	b, err := os.ReadFile(expectedDownloadedFilePath)
+	require.NoError(t, err)
+
+	contents := string(b)
+	assert.Contains(t, contents, "apiVersion: v1")
+	assert.Contains(t, contents, "name: my-nginx-svc")
+	assert.Contains(t, contents, "type: LoadBalancer")
+	assert.Contains(t, contents, "image: nginx:1.14.2")
+}
+
+func TestConfigureKubernetes_SuccessfulRKE2ServerWithManifests(t *testing.T) {
+	ctx, teardown := setupContext(t)
+	defer teardown()
+
+	ctx.ImageDefinition.Kubernetes = image.Kubernetes{
+		Version: "v1.29.0+rke2r1",
+		Network: image.Network{
+			APIVIP:  "192.168.122.100",
+			APIHost: "api.cluster01.hosted.on.edge.suse.com",
+		},
+	}
+	ctx.KubernetesScriptInstaller = mockKubernetesScriptInstaller{
+		installScript: func(distribution, sourcePath, destPath string) error {
+			return nil
+		},
+	}
+	ctx.KubernetesArtefactDownloader = mockKubernetesArtefactDownloader{
+		downloadArtefacts: func(arch image.Arch, version, cni string, multusEnabled bool, destPath string) (string, string, error) {
+			return serverInstaller, serverImages, nil
+		},
+	}
+
+	ctx.ImageDefinition.Kubernetes.Manifests.URLs = []string{
+		"https://k8s.io/examples/application/nginx-app.yaml",
+	}
+
+	k8sCombDir := filepath.Join(ctx.CombustionDir, k8sDir)
+	require.NoError(t, os.Mkdir(k8sCombDir, os.ModePerm))
+
+	localManifestsSrcDir := filepath.Join(ctx.ImageConfigDir, k8sDir, manifestsDir)
+	require.NoError(t, os.MkdirAll(localManifestsSrcDir, 0o755))
+
+	localSampleManifestPath1 := filepath.Join("..", "registry", "testdata", "sample-crd.yaml")
+	err := fileio.CopyFile(localSampleManifestPath1, filepath.Join(localManifestsSrcDir, "sample-crd.yaml"), fileio.NonExecutablePerms)
+	require.NoError(t, err)
+
+	localSampleManifestPath2 := filepath.Join("..", "registry", "testdata", "invalid-crd.yml")
```

Comment:
Something we discussed last week is that we don't particularly want to do this since it suggests that we're able to ship invalid manifests. Can we drop it?